src/util/side_metadata/helpers_32.rs

```diff
@@ -165,8 +178,8 @@ pub fn try_mmap_metadata_chunk(
     let policy_meta_start = address_to_meta_chunk_addr(start);
     if !no_reserve {
         // We have reserved the memory
-        unsafe { memory::dzmmap(policy_meta_start, local_per_chunk) }
+        MMAPPER.ensure_mapped(policy_meta_start, local_per_chunk >> LOG_BYTES_IN_PAGE)
     } else {
-        memory::mmap_noreserve(policy_meta_start, local_per_chunk)
+        MMAPPER.quarantine_address_range(policy_meta_start, local_per_chunk >> LOG_BYTES_IN_PAGE)
```

Comment:
What if `local_per_chunk` is less than one page?
E.g. 1-bit per chunk?

Reply:
`local_per_chunk` here is the bytes needed for the chunk, right? We should align up - I will fix this. But I don't understand your example of 1-bit per chunk. Do I misunderstand anything?

Reply:
Right, it is the number of side metadata bytes per chunk.
My example is the case where we want n-bit of metadata per chunk (e.g. to store chunk mapping state).
In that case, we only need n-bit per chunk, and if we mmap an aligned up size of 1 page for it, we will re-mmap the same page for the future chunks

Reply:
The mmapper ensures that won't happen. If mmapper finds the chunk is quarantined or mapped, it will not do anything. This is exactly the reason for this PR.

Reply:
I meant the mmapper ensures we won't re-mmap the same page, as it knows the state of each chunk.